lib/module.js

```diff
@@ -152,14 +193,18 @@ export default function SentryModule (moduleOptions) {
         Sentry.captureException(error)
       }))
     })
-  } else {
-    this.addPlugin({
-      src: resolve(__dirname, 'sentry.mocked.js'),
-      fileName: 'sentry.server.js',
-      mode: 'server'
-    })
   }
 
+  this.addPlugin({
+    src: resolve(__dirname, `plugin.${pluginOptionServer}.js`),
+    fileName: 'sentry.server.js',
+    mode: 'server',
+    options: {
+      dev: this.options.dev,
```

Comment:
I think it shouldn't be necessary to pass `dev` to server.

Reply:
I think I added this to only log warnings in development mode but forgot to use it in the server plugin. Not sure if its a good idea or not, but I dont think you'd ever want to log `sentry.${key}() called, but Sentry plugin is disabled. Arguments` in production...

Reply:
But then `logMockCalls` is used for deciding that so users can have any logic they want.

Reply:
Yeah, thats true. Just feels as a bad practice we should prevent to log such a warning message in prod. But am ok with leaving it as is